### ISSUE TITLE
command line ooptions like windbg or gdb

### ISSUE THREAD

**Issue by Byzero512** [x64dbg/x64dbg#2201]
please add command line options to x64dbg, so that  it can insert breaskpoints or exec commands when attach to a process use Cmd, just like windbg:

![image](https://user-images.githubusercontent.com/36794026/61791442-98740a80-ae4c-11e9-84db-b21bdb4dc2ef.png)

https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/windbg-command-line-options

**Comment by mrexodia** [comment]
The problem with things like this is that it goes on indefinitely feature-wise. What is not yet covered by the debuggee initialization script and the existing command line to load an executable in the debugger?

**Comment by Byzero512** [comment]
well, I think the two import things are:
1. write commands in a file, and when init the dwbuger, it will exec the comand. like gdb's .gdbinit.
2. provide a cmd that like "shell" on gdb to exec shell command.
maybe reference gdb though the picture posted is of windbg.

**Comment by mrexodia** [comment]
Sure and what would be your use case for arbitrary programs? There already exists a feature to execute your commands when a specific debuggee is being initialized.

With regards to a shell there are no plans for any command line support, but I did implement a prototype that I will open source. If you want something more specialized it is definitely possible to write it. The executable `x64dbg.exe` is a **very** thin wrapper around `x64dbg.dll` and you can easily reimplement it as a console application with all the command line capabilities you need (for example by passing data to a plugin you write).

If it turns out that your custom solution is useful to the general public I will definitely consider implementing it upstream, but x64dbg is not gdb and will never be because it's designed with a very different philosophy.

**Comment by Byzero512** [comment]
well,I use these feature because sometimes I will interactive with the debuggee with python, and use python subprocess module to let X64dbg attach to the debuggee.
and I want to insert some breakpoint when init the debugger that need cms options.
as you mention, debugger can execute command when init, but I do not found this feature.

**Comment by mrexodia** [comment]
You can find the documentation here: https://help.x64dbg.com/en/latest/gui/menus/Options.html?highlight=initialization#set-initialization-script

**Comment by Byzero512** [comment]
thanks.

**Comment by noword** [comment]
I wrote a program, it can run x64dbg in the console.
It's basically useless without script plugin.
Hope it can help someone.

https://github.com/noword/x64dbg_console

**Comment by mrexodia** [comment]
Nice! There’s also https://github.com/x64dbg/x64dbg_headless if you’re interested.

**Comment by noword** [comment]
> Nice! There’s also https://github.com/x64dbg/x64dbg_headless if you’re interested.

I hook all GuiXXXX functions in x64bridge.dll.
This reminds me，just hook 3 functions in x64gui.dll is enough.
Thank you.

**Comment by mrexodia** [comment]
Yeah, like I wrote in this thread: if this becomes useful for more people I’m willing to integrate it into x64dbg itself.

One example I can think of is to use this for headless x64dbg tests. Currently there are no tests at all and it could be useful to test different combinations of breakpoints and settings…
